tools/isobuild/package-api.js

```diff
@@ -339,13 +339,39 @@ _.extend(PackageAPI.prototype, {
     var errors = [];
     _.each(paths, function (path) {
       forAllMatchingArchs(arch, function (a) {
+        const isAsset = !! (fileOptions && fileOptions.isAsset);
+
+        // Check if we have already added a file at this path
         if (_.has(self.sources[a], path)) {
-          errors.push("Duplicate source file: " + path);
+          const otherSource = self.sources[a][path];
+
+          // Files can actually be added twice for each architecture: once as
+          // an asset, and once as a source file
+          const otherFileOptions = otherSource.fileOptions;
+          const otherFileIsAsset = !! (otherFileOptions &&
```

Comment:
Can't you just do that with eg

```
api.addFiles("foo.js", "client");
api.addFiles("foo.js", "server", {isAsset: "both"});
```

I'm not saying this is the best/only answer, but I'm not sure that what you're talking about isn't possible with explicit-both.